modules/internal/ChapelRange.chpl

```diff
@@ -1778,25 +1778,45 @@ private proc isBCPindex(type t) param do
       if _low > _high then return false;
     }
 
-    if ! this.isAligned() || ! other.isAligned()
+    if ! this.isAligned()
       then return false;
 
     return true;
   }
 
   inline proc range.chpl_boundsCheck(other: range(?e,?b,?s))
   {
-    if ! this.isAligned() || ! other.isAligned()
+    if ! this.isAligned()
       then return false;
 
-    var boundedOther = new range(
+    var boundedOther = if other.isAligned() then
+                        new range(
                           idxType, boundKind.both, other.strides,
                           if other.hasLowBound() then other._low else _low,
                           if other.hasHighBound() then other._high else _high,
-                          other._stride, other._alignment);
+                          other._stride, other._alignment)
+                      else if this.hasPosNegUnitStride() then
```

Comment:
I suggest to add the following comment, or similar, before the above else-clause:

If 'other' is not aligned, we want to create an aligned range to pass it to contains() below, because otherwise it will always return false. For that, we copy alignment from 'this'. If 'this' is unit-stride, the alignment is zero, otherwise this.alignment potentially needs to be normalized w.r.t. other.stride.